Description: Fix afpd segfault in Spotlight SPARQL parser
Author: Andrew Stormont <andyjstormont@gmail.com>
Bug: https://github.com/Netatalk/Netatalk/issues/56
Bug: https://github.com/Netatalk/Netatalk/pull/57
Last-Update: 2020-09-25
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/etc/spotlight/sparql_parser.y
+++ b/etc/spotlight/sparql_parser.y
@@ -119,6 +119,8 @@
         yyerror("Spotlight queries with logic expressions are disabled");
         YYABORT;
     }
+    if ($1 == NULL || $3 == NULL)
+        YYABORT;
     if (strcmp($1, $3) != 0)
         $$ = talloc_asprintf(ssp_slq, "{ %s } UNION { %s }", $1, $3);
     else
